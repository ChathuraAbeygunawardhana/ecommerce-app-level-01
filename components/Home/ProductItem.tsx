import React from 'react';
import { StyleSheet, Text, View, Image, Pressable } from 'react-native';

const itemWidth = 220;

interface ProductItemProps {
  item: {
    mainImage: string;
    name: string;
    price: number;
  };
  onPress: () => void;
}

const ProductItem: React.FC<ProductItemProps> = ({ item, onPress }) => (
  <Pressable style={styles.itemContainer} onPress={onPress}>
    <View style={styles.imageContainer}>
      <Image source={{ uri: item.mainImage }} style={styles.image} />
    </View>
    <Text style={styles.name} numberOfLines={1}>
      {item.name}
    </Text>
    <Text style={styles.price}>${item.price}</Text>
  </Pressable>
);

export default ProductItem;

const styles = StyleSheet.create({
  itemContainer: {
    flex: 1,
    marginBottom: 15,
    marginHorizontal: 5,
    padding: 8,
    backgroundColor: '#fff',
    borderRadius: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    width: '100%',
    height: 180,
    borderRadius: 10,
    resizeMode: 'cover',
  },
  name: {
    fontSize: 14,
    marginTop: 8,
    fontWeight: '500',
  },
  price: {
    fontSize: 14,
    marginTop: 4,
    color: '#666',
  },
});
