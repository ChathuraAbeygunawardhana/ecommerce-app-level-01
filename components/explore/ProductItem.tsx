import React from 'react';
import { View, Text, Image, TouchableOpacity, StyleSheet, Dimensions } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Colors } from '../../constants/Colors';
import { useTheme } from '../../contexts/ThemeContext';
import { useFavourites } from '../../contexts/FavouritesContext';

interface ProductItemProps {
  item: any;
  onPress: () => void;
}

const ProductItem: React.FC<ProductItemProps> = ({ item, onPress }) => {
  const { theme } = useTheme();
  const currentColors = Colors[theme as 'light' | 'dark'];
  const { addToFavourites, removeFromFavourites, isFavourite } = useFavourites();

  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          styles.itemContainer,
          { backgroundColor: currentColors.background_02, padding: 10 },
        ]}
      >
        <View style={styles.imageContainer}>
          <Image source={{ uri: item.mainImage }} style={styles.image} />
        </View>
        <Text style={[styles.name, { color: currentColors.text }]}>
          {item.name}
        </Text>
        <Text style={[styles.price, { color: currentColors.grey }]}>
          ${item.price}
        </Text>
        <View style={{ height: 10 }} />
        <View style={styles.addIconContainer}>
          <TouchableOpacity
            style={[styles.addIcon, { backgroundColor: currentColors.lightBlue }]}
            onPress={() =>
              isFavourite(item.id)
                ? removeFromFavourites(item.id)
                : addToFavourites(item)
            }
          >
            <Ionicons
              name={isFavourite(item.id) ? 'heart' : 'add'}
              size={20}
              color="white"
            />
          </TouchableOpacity>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  itemContainer: {
    width: (Dimensions.get('window').width - 10 * (2 + 1)) / 2,
    marginBottom: 15,
    borderRadius: 20,
    overflow: 'hidden',
  },
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    width: '90%',
    height: 130,
    borderRadius: 10,
    resizeMode: 'cover',
    margin: 10,
  },
  name: {
    marginTop: 2,
    fontSize: 14,
    color: Colors.light.text,
  },
  price: {
    fontSize: 14,
    marginTop: 5,
    color: Colors.grey,
  },
  addIconContainer: {
    position: 'absolute',
    bottom: -10,
    right: -10,
    borderRadius: 10,
    paddingRight: 10,
    paddingBottom: 10,
    paddingTop: 5,
    paddingLeft: 5,
    backgroundColor: Colors.lightBlue,
  },
  addIcon: {
    borderRadius: 10,
    paddingRight: 10,
    paddingBottom: 10,
    paddingTop: 5,
    paddingLeft: 5,
  },
});

export default ProductItem;
