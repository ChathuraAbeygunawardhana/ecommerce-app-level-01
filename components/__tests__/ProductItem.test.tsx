
import React from 'react';
import renderer from 'react-test-renderer';
import ProductItem from '../Home/ProductItem';

describe('ProductItem', () => {
  it('renders correctly', () => {
    const item = {
      mainImage: 'https://example.com/image.jpg',
      name: 'Product Name',
      price: 100,
    };
    const tree = renderer.create(<ProductItem item={item} onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});