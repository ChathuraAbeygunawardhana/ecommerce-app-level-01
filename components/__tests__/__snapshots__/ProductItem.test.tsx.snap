// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "flex": 1,
      "marginBottom": 15,
      "marginHorizontal": 5,
      "padding": 8,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 180,
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      {
        "fontSize": 14,
        "fontWeight": "500",
        "marginTop": 8,
      }
    }
  >
    Product Name
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
        "marginTop": 4,
      }
    }
  >
    $
    100
  </Text>
</View>
`;
